### ISSUE TITLE
Request: Please add chocolatay installation for flux2 for Windows

### ISSUE THREAD

**Issue by joerg-walter-de** [fluxcd/flux2#1440]
Please add chocolatay installation for flux2 for Windows

**Comment by kingdonb** [comment]
I don't know about adding chocolatey support myself, but `gofish` has an installer that is kept updated with Flux v2 releases

It works the same in Windows, Mac, and Linux, across all supported architectures:

https://gofi.sh/#install

```
$ gofish install flux
==> Installing flux...
🐠  flux 0.13.4: installed in 525.801972ms
```

**Comment by stefanprodan** [comment]
I've asked @JimPruitt who maintains the Flux v1 CLI package on Chocolatay if he can add the v2 CLI. https://github.com/JimPruitt/chocolatey-packages/issues/6

**Comment by JimPruitt** [comment]
Just a follow up here.  Flux2 has been submitted to Chocolatey.org and it is currently pending moderation.  https://community.chocolatey.org/packages/flux/0.16.0 .  I'll look further into automation that will respond to new releases as they become available.

**Comment by stefanprodan** [comment]
@JimPruitt can you please update Flux description like here https://operatorhub.io/operator/flux Thanks!

PS. Flux is no longer a Weaveworks project, it has been donated to Linux Foundation (CNCF). Please change the tittle to “Flux CLI”.

**Comment by JimPruitt** [comment]
Will do.  It’s hung up in moderation right now.   I’ll see if I can’t move this along.

**Comment by JimPruitt** [comment]
@stefanprodan Just a heads up,  I had made your requested changes and attempted to update the Flux2 chocolatey package to version 0.16.1.  Unfortunately, I'm not allowed to submit the 0.16.1 update until the chocolatey.org moderators manually approve version 0.16.0.  

It appears that one of the virus checks (Cylance) flagged the zipped release as 'suspect' (t really has no issues at all - but this is typical of Cylance).  I recall that this issue had occurred with the original 'fluxctl' submission as well.   The best course is to wait until the moderators manually approve it.

Once this is approved, I will submit version 0.16.1 with your updates.

**Comment by JimPruitt** [comment]
@stefanprodan  Just another follow up here, I am still seeing an issue with VirusTotal (specifically Cylance) flagging the flux2 windows amd64 release as suspect (for the 0.16.0 and 0.16.1 releases).  This is holding up the automated moderation within Chocolatey for the past 13 days now.  The package is still pending moderation and is #43 in the moderation queue currently, but movement has been extremely slow despite my efforts to 'bump' the issue.

I believe that Cylance (by default) flags new .exe's as 'bad' until it's manually authorized.  This article explains more https://www.csoonline.com/article/3216765/heres-why-the-scanners-on-virustotal-flagged-hello-world-as-harmful.html .  
I will try to work with Cylance as well to see if I can submit a 'false positive' appeal.

**Comment by kingdonb** [comment]
I applaud your contribution here and I don't want to belabor this as it seems you're already on top of it.

My hope is that they will use automation to slurp up Flux releases, and proactively allow-list them. I'm sure they (Cylance or VirusTotal) will be helpful but I also see a risk and want to help if possible to see this doesn't become a new time sink for us.

This kind of idea (release acceptance automation) is exactly what I believe packaging [platforms like GoFish are doing](https://github.com/fishworks/fish-food/issues/249), and IMHO it's exactly why their releases can stay fairly current even if they are maybe not using Flux, with a person in the loop who lends their direct approval to each new release. Even with that, from what I understand this works out to only a modest outlay of direct effort.

(And if we're in luck that sort of approach will be in Cylance's response! I don't want to sound pessimistic but I won't be surprised if the reply is less positive than that.)

@JimPruitt I am trying to avoid a wall of text, and it is difficult for me, but first I want to reiterate that I really appreciate you're spending your time on this with us. To get to the point, my ideal outcome is not just that we don't come away with a new recurring task for every release, but ideally that nobody working on a CNCF project ever has to revisit this issue again unless for some reason in particular they "do not fit a mold." 

The CNCF sponsors security audits for many projects, and including Flux, projects that are in Incubation status before attaining Graduated status are required to undergo such a security audit. The CNCF, I believe, are proactive about making sure that projects like ours and others on the landscape, which they list and recommend, are good actors (and definitely not involved in developing malware for threat-actors.)

I know that's over-simplifying a bit, but maybe that argument will pull some weight with the scanning vendors.

The CNCF vouch should count for something, and certainly every CNCF project cannot be expected to engage all of these companies and spend project members time to get a separate review and get each new release added to their allow lists.

I will be glad if the issue is resolved for Flux, and I don't want to make extra work for you or anyone. I just wanted to point out that there is automation possible so this does not turn into a recurring task. Thanks again for finding the root cause.

**Comment by kingdonb** [comment]
I've heard it said by the maintainers that "publishing a catalog with stale Flux releases is worse than not publishing any release at all" with respect to offerings like this one and like our OperatorHub catalog. [That catalog entry](https://operatorhub.io/operator/flux) for Flux was lagging behind for a while, then someone was able to pick it up and design a process to support it permanently. Some automation was implemented, and now we're glad to say that our OperatorHub entry keeps up with our release pace.

I know that is not the whole story though, there remains some manual work to keep that stream current as well as testing, and for members of our community this can turn into a burden, frankly release engineering is hard enough without "one more straw." We are part of a major public benefit project (CNCF) who exists because of the graceful contributions of our members and contributors, and by contributions I mean time spent more than anything else, our most precious commodity.

**Comment by JimPruitt** [comment]
@kingdonb I can appreciate your response and  concerns over this.  I'm not particularly enamored over the fact that I have very little recourse here in light of the Chocolatey.org moderation practices.  From what I understand they leverage third party virus scanning resources that choose 'nontrust' by default.  From what I'm seeing, there is very little movement in the Chocolaty moderators response.  Maintenance of this would always require  some level of manual intervention going forward.

I will leave this package 'as-is' in the hopes that it will eventually get moved along.  You documenting 'gofish' as the preferred install method for Windows is probably your best course of action here.

**Comment by JimPruitt** [comment]
Over the past weekend, the 'Flux' chocolatey package (0.16.0)  had received approval and is now available.  I have quickly submitted version 0.16.1 with the requested formatting changes to the 'readme'.

**Comment by stefanprodan** [comment]
@JimPruitt thanks for getting flux v2 on chocolatey. Can you please change the title as this is super confusing: 

<img width="1140" alt="Screenshot 2021-07-26 at 16 08 07" src="https://user-images.githubusercontent.com/3797675/126993905-b9e67883-d773-4c43-b807-f765e629aa07.png">

In our website we refer to Flux v1 as `legacy`, so I would rename the v1 to `Flux legacy CLI`, and the new package to `Flux v2 CLI`.

**Comment by hiddeco** [comment]
In general, any reference to Weaveworks can be omitted (except where part of the project history is explained), as we have been a CNCF project for quite awhile now.

**Comment by JimPruitt** [comment]
Sure thing @stefanprodan.  Tonight I'll submit Flux v1 (version 1.23.1) with your suggested title change to "Flux legacy CLI" which should help delineate the two.   Note also that Flux v2 (version 0.16.0) was approved. After that submission was sent,  you had requested additional changes to the description format which are still pending.   I have submitted Flux v2 (version 0.16.1) with your suggested description changes from above to make (the) "Flux description like here https://operatorhub.io/operator/flux" .  Once this is approved that format change will be visible in the description (right now it looks just like the legacy description because I didn't know any better at the time of original submission).

@hiddeco , We are talking about 2 separate chocolatey packages (Flux v1, Flux v2) here and it's terribly confusing to try and discern if you are talking about one or the other. It would help out a lot if you can provide me an explicit example of what changes you are after specifically.  If you desire changes to be made to the description on Flux v1, let's track those requests in a separate issue here https://github.com/JimPruitt/chocolatey-packages/issues.   In general, I'm happy to tailor the package descriptions to any spec you folks want to see, just provide me the correct markdown and I'll update it.

**Comment by kingdonb** [comment]
I think it's the "weaveworks" tag that could be removed from both, I'm guessing that's what is being suggested – though Flux's genesis was at weaveworks, the entire FluxCD org is a separate entity in the CNCF since its donation some time ago.

Any mention of `Weaveworks Flux` should be scrubbed and the link to https://github.com/weaveworks/flux could be fixed up to point to the fluxcd/flux org repo instead, or simply "Flux" - with respect to Fluxctl, all of that I mean. I haven't reviewed both except to note that searching for "flux" seems to cause the chocolatey servers to struggle and strain a bit. At first it returned no results at all, then searching for "fluxctl" found both packages, maybe it had something to do with caches being populated for the first time because now it seems to return a response within not too many seconds.

I'll submit a PR to your chocolatey-packages repos as you suggested 👍 thanks again for this.

**Comment by stefanprodan** [comment]
Why aren’t we using the amd64 build instead of this https://github.com/JimPruitt/chocolatey-packages/blob/master/flux2/tools/chocolateyinstall.ps1#L8?

**Comment by JimPruitt** [comment]
Hrmm....   that's good point, and the answer is "You're right, there isn't a really good reason not to support 64 bit architectures where applicable".   

In order to do so, I need to make an enhancement to the package to autodetect the current architecture and automatically select the correct executable.   I think I can also get it to support manual overrides.    I'll look into this today...

**Comment by JimPruitt** [comment]
@kingdonb  The "Weavworks" tag was previously in the "<tags>" section of the .nuspec file on both Flux1 and Flux2.  I had removed it with a previous release and therefore you couldn't find it in the Master branch.  This push had not yet cleared Chocolatey moderation so it might not yet be visible on Chocolately.org.  (that painful 2 week 'lag' between build and moderation approval.)  Once the Flux2 package gets 'trusted' (which usually takes a bit longer) we will start seeing faster turnaround times for this.

**Comment by stefanprodan** [comment]
I see that https://community.chocolatey.org/packages/flux has been updated to latest version super fast. Thank you very much @JimPruitt 🎖️ 🏅 🥇
